crates/primitives/src/deposit.rs

```diff
@@ -147,12 +147,19 @@ impl DepositInfo {
         })
     }
 
-    fn compute_spend_infos(
+    /// Computes the spend info for the Deposit Request Transaction (DRT).
+    ///
+    /// This also performs the following validations:
+    ///
+    /// 1. The script pubkey in the DRT matches the one provided in the [`DepositInfo`].
+    /// 1. The x-only public key in the DRT matches the one provided in the [`DepositInfo`].
+    /// 1. The script pubkey in the DRT is a valid taproot script.
```

Comment:
This treats the `DepositInfo` and the `DRT` as two separate things but they really are two representations of the same underlying thing. I'd update the doc to something like the following:

```suggestion
    /// 1. The taproot address computed from the x-only public key in the DRT and the bridge multisig address is the same as the output address in the DRT.
    /// 1. The control block computed from the DRT commits to the multisig script that the Deposit Transaction (DT) spends.
```